### DISCUSSION TITLE
ListBox Event when items are binded

### DISCUSSION THREAD

**Discussion by Tenjim** [AvaloniaUI/Avalonia#9097] [Q&A] [answered]
Hi guys, 

Just a question, there is an event to know when ListBoxItems on a Listbox are binded ?
I'm struggling to subscribe when ListBoxItems is loaded.

Thanks in advance

**Comment by timunie** [comment, marked as answer]
You can observe items property. The real question is, what is your goal? Most of the times the logic should be handled by the ViewModel

**Reply by Tenjim** [reply]
to Add EventHandler on ListboxItem to doing drag and drop. On Avalonia sample, it's just two static Textblock to drag and drop.

**Reply by timunie** [reply]
In that case check out Avalonia.Behaviors repo. They support this out of the box

**Reply by Tenjim** [reply]
Yeah I saw that on @wieslawsoltes repos, but I wanted to use your native drag and drop :)

Thanks for reply btw :)

**Reply by timunie** [reply]
You can also write your own behavior in that case. But keep in mind, ListBos is virtualized. That means items can use the same container. They can also be removed during drag if you don't pay attention to it. So if the given lib works, i would not do the effort to reinvent the wheel.
